cfme/scripting/report_parser.py

```diff
@@ -0,0 +1,97 @@
+"""Script to parse test reports."""
+import json
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+import click
+import yaml
+
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+
+# Log to stdout too
+add_stdout_handler(logger)
+
+
+class ReportParser:
+    def __init__(self, report_path, report_format='json', report_dict=None, *args, **kwargs):
+        self.report_path = report_path
+        self.report_format = report_format  # for now only support json
+        self.report_dict = report_dict or self._pull_in_report()
+
+    def _pull_in_report(self):
+        if self.report_format == 'json':
+            try:
+                with open(self.report_path) as json_file:
+                    return json.load(json_file)
+            except IOError:
+                logger.exception("File {} does not exist".format(self.report_path))
+                raise
+
+    def parse_report_per_user(self, filter_by='failed', user=''):
+        tests = self.report_dict['tests']
+        assignee_wise_results = {}
+        for test_case in tests:
+            try:
+                result = tests[test_case]['statuses']['overall']
+            except KeyError:
+                # continue loop if any test is missing the statuses/overall
+                continue
+            # if user='' it will fetch for all users
+            if (result == filter_by and user in tests[test_case]['assignee']):
+                try:
+                    assignee_wise_results[tests[test_case]['assignee']].update(
+                        {test_case: result})
+                except KeyError:
+                    assignee_wise_results[tests[test_case]['assignee']] = (
+                        {test_case: result})
+        return assignee_wise_results
+
+
+@click.group("blame", help="Functions for generating reports on Tests grouped by user")
+def main():
+    pass
+
+
+@main.command("test-report", help="Returns a Dictionary containing failed test cases per user")
```

Comment:
Thoughts on naming this something else? My concern is that this makes it sound like you're running a test. One thought is that the calling sequence could be `miq blame <user>` so that it makes it seem like you're literally blaming the person :smile: The all the other args could be optional. Not sure if it's worth the refactor though, so I'd like to hear your thoughts.

Reply:
@john-dupuy I'd like to get one more review on this before I refactor. Don't want to refactor only to go back to what it was. I like your idea too. :)